lib/glimesh/channel_lookups.ex

```diff
@@ -113,6 +113,62 @@ defmodule Glimesh.ChannelLookups do
     |> Repo.preload([:category, :user, :stream, :subcategory, :tags])
   end
 
+  def count_live_followed_channels_that_are_hosting(%User{} = user) do
+    from(c in Channel,
+      left_join: f in Follower,
+      on: c.user_id == f.streamer_id,
+      join: ch in ChannelHosts,
+      on: c.id == ch.hosting_channel_id,
+      join: target in Channel,
+      on: target.id == ch.target_channel_id,
+      left_join: target_followers in Follower,
+      on: target.user_id == target_followers.streamer_id,
+      where: ch.status == "hosting",
+      where: target.status == "live",
+      where: f.user_id == ^user.id,
+      where: target_followers.user_id != ^user.id or is_nil(target_followers.user_id),
+      distinct: target.id,
+      select: [target]
+    )
+    |> Repo.all()
+    |> length()
+  end
+
```

Comment:
I've noted the following method in my "Performance Considerations" section.  This query mostly utilizes joins on indexes but I have not been able to test it with a large dataset.